pkg/translate/translate.go

```diff
@@ -337,85 +358,58 @@ func (t *Translator) ValuesOverlaysToHelmValues(in map[string]interface{}, cname
 	return out
 }
 
-// ToComponent returns the Components under the featureName feature.
-// TODO: should do this once only, create a NewTranslator and do it there.
-func (t *Translator) ToComponent(featureName name.FeatureName) []name.ComponentName {
-	var out []name.ComponentName
-	for c, f := range t.ToFeature {
-		if f == featureName {
-			out = append(out, c)
-		}
-	}
-	return out
+// Components returns the Components under the featureName feature.
+func (t *Translator) Components(featureName name.FeatureName) []name.ComponentName {
+	return t.featureToComponents[featureName]
 }
 
-// protoToValues takes an interface which must be a struct ptr and recursively iterates through all its fields.
+// protoToHelmValues takes an interface which must be a struct ptr and recursively iterates through all its fields.
 // For each leaf, if looks for a mapping from the struct data path to the corresponding YAML path and if one is
 // found, it calls the associated mapping function if one is defined to populate the values YAML path.
 // If no mapping function is defined, it uses the default mapping function.
-func (t *Translator) protoToValues(structPtr interface{}, root map[string]interface{}, path util.Path) (errs util.Errors) {
-	dbgPrint("protoToValues with path %s, %v (%T)", path, structPtr, structPtr)
-	if structPtr == nil {
+func (t *Translator) protoToHelmValues(node interface{}, root map[string]interface{}, path util.Path) (errs util.Errors) {
+	dbgPrint("protoToHelmValues with path %s, %v (%T)", path, node, node)
+	if node == nil {
```

Comment:
FYI nil interface checks are tricky (see https://play.golang.org/p/4gRVhIEVT9C). Specifically, this might evaluate to non-nil for a nil pointer (typed nil interface).

Reply:
Yeah, that's why I created util.IsValueNil - and subsequently forgot to use it here :-)
I've corrected this problem in https://github.com/istio/operator/pull/81